numpy/lib/function_base.py

```diff
@@ -1339,18 +1339,18 @@ def angle(z, deg=0):
     45.0
 
     """
-    if deg:
-        fact = 180/pi
-    else:
-        fact = 1.0
-    z = asarray(z)
-    if (issubclass(z.dtype.type, _nx.complexfloating)):
+    z = asanyarray(z)
+    if issubclass(z.dtype.type, _nx.complexfloating):
         zimag = z.imag
         zreal = z.real
     else:
         zimag = 0
         zreal = z
```

Comment:
perhaps worth noting that the code path where z isn't a subclass of `complexfloating` doesn't have test coverage, although that's probably not an issue with changes made here